src/std/net/socket/server-test.ss

```diff
@@ -0,0 +1,84 @@
+;;; -*- Gerbil -*-
+;;; (C) vyzo at hackzen.org
+;;; Socket server tests
+
+(import :gerbil/gambit/threads
+        :std/misc/threads
+        :std/net/socket
+        :std/net/socket/base
+        :std/os/socket
+        :std/error
+        :std/sugar
+        :std/test)
+(export socket-server-test)
+
+(def socket-server-test
+  (test-suite "test :std/net/socket"
+
+    (def server-address "localhost:4242")
+
+    (def (echo sock)
+      (def buf (make-u8vector 256))
+      (try
+       (let lp ()
+         (def len (ssocket-recv sock buf))
+         (if (= len 0)
+           (begin
+             (ssocket-close sock)
+             (values))
+           (begin
+             (ssocket-send sock buf 0 len)
+             (lp))))
+       (catch (io-error? exn)
+         (values))))
+
+    (def (run-echo-server address)
+      (def sock (ssocket-listen address))
+      (let lp ()
+        (spawn/group 'echo-server echo (ssocket-accept sock))
```

Comment:
You are not quite using `spawn/group` correctly -- you should use it in the top level thread (here the echo server) and then use plain spawn for the workers.
Here you end up creating a new thread group for every worker, while I believe the intention is to create a thread group for the entire echo server and its workers.

Reply:
Yes, that was indeed the intention, I think I just misunderstood the behavior of `spawn/group`. I'll change things accordingly and generally try to make this a little more robust, maybe using the httpd code as a guide.

Reply:
All you have to change here is use `spawn` or `spawn/name` for the echo thread :)